### PR TITLE
Improve testing and testing environment

### PR DESCRIPTION
**Pull request by CoMPaTech**
Introduce ability to py3-virtualenv locally and straighten out the tests/README instructions as well.

### CODE REVIEW

**Comment by CoMPaTech** [comment]
Ok, this one is ready for review:
As indicated on discord, test-suite has (largely due to full indentation) a very large diff now.
Changes to Smile.py are through black - changing of the docstrings as per checks.
As for pylint/flake8/pydocsstyle, theese aren't included in Travis (yet) - but are part of the local tests. All files involved are now compliant.

**Comment by CoMPaTech** [comment]
No version bump present (yet) as there are no (real) codechanges, but can be added.

**Comment by CoMPaTech** [comment]
Confirmed full-compliance can be ran - undone for now

File: tests/README.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # To be detailed
 
-Basically make sure you `pip -r requirements.txt install`
+Ensure you have python(3) with virtualenv installed.
+From the main directory of this repository run `python3 -m venv venv` followed by `source venv/bin/activate`
 
-Tests should be included in `.travis.yml` or `.github/actions/` later on but you can also run them
+Now prepare your virtual environment by runnin `pip install -r tests/requirements.txt -r requirements`
```
Comment (bouwew) [inline tests/README.md]: running

**Pull request merged**